### DISCUSSION TITLE
Getting logs for microcks users

### DISCUSSION THREAD

**Discussion by ymeudal** [microcks/microcks#874] [Ideas]
When microcks is deployed in a cloud environment or in a standalone server, it is impossible for microcks users to view logs without an external connection to console logs; this external connection is unthinkable in a prod infrastructure.
Most interesting things to be able to view are logs issued from mock requests: 
- content of what is written in "log" object from a script,
- getting exceptions during a script execution (at runtime or compilation errors),
- Obtaining what response is selected by any type of dispatcher (for example, the string returned by script)

I would consider one way to do it is to get logs when the request contains a special query parameter, like a ?microcks_debug  after the URI. Microck would return a text/plain content of debug logs (including the string reference of the selected response) instead of the selected response.

**Comment by lbroudoux** [comment]
Hi @ymeudal,

Thanks for the suggestion. I think about it and see it analogous to an "explain plan" feature on a database: you'll get the information, logs, and decisions made by the mocking engine with actual reference to the response that would be returned. Maybe what we'd need now is a structured way of presenting this. It could be an array of log messages only OR some more accurate structure presenting decisions ... Any idea or suggestion on this?

**Comment by ymeudal** [comment]
Hello @lbroudoux,
For my purposes, I do not have to parse this debugging response, so the response may be a plain text.
Structuring the response with elements splitted in a json will be better if the body is processed by a checking script.
